tensorflow/java/src/main/java/org/tensorflow/Output.java

```diff
@@ -57,6 +60,30 @@ public Output asOutput() {
     return this;
   }
 
+  @Override
+  public int hashCode() {
+    return operation.hashCode() + 37 * index + 17;
```

Comment:
According to the stackoverflow thread you linked to `Objects.hash` implements the same approach (though, looking at the sources I'm not sure that that's true).

Anyway, I figured `Objects.hash` should encode the best practice :)